catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/ResourceCache.java

```diff
@@ -289,15 +291,17 @@ public void addPendingCacheEntry(ReliableResource reliableResource) {
     /**
      * @param key
      * @return Resource, {@code null} if not found.
+     * @throws CacheException if the key or latestMetacard are null, or when a bad product is found
+     * in the cache directory.
      */
     @Override
-    public Resource getValid(String key, Metacard latestMetacard) {
+    public Resource getValid(String key, Metacard latestMetacard) throws CacheException {
         LOGGER.debug("ENTERING: get()");
         if (key == null) {
-            throw new IllegalArgumentException("Must specify non-null key");
+            throw new CacheException("Must specify non-null key");
         }
```

Comment:
throwing custom checked exceptions for violations of method contracts is going to make for a lot of custom exceptions and try/catch blocks.  If the client is calling the method correctly (with non-null arguments), then they'll never see these.  I think IllegalArgumentException is appropriate here.